### ISSUE TITLE
Crash in Terraform Console

### ISSUE THREAD

**Issue by tombuildsstuff** [hashicorp/terraform#17867]
### Terraform Version
```
$ terraform --version
Terraform v0.11.8-dev (05291ab9822dd794b8ad61a902af832898618d99+CHANGES)
```

### Terraform Configuration Files
```
$ terraform console
> azurerm_resource_group.test.name
```

### Crash Output
https://gist.github.com/tombuildsstuff/ecb051924409d40a682e91b36c4ed8ab

### Expected Behavior
Terraform should have stated the resource couldn't be found

### Actual Behavior
Terraform Crashed

### Additional Context
I ran Terraform Console in a folder without an associated Statefile (unintentionally), and then queried for a resource. Whilst I ran Terraform in the wrong folder - I'd expect an error message rather than a crash

**Comment by atz** [comment]
Same error here.  At the very least, terraform should not segfault. 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1755a04]
```

**Comment by apparentlymart** [comment]
I just tested this in v0.12.0-alpha4 to confirm, and indeed it is fixed:

```
$ terraform console
> azurerm_resource_group.test.name

Error: Reference to undeclared resource

  on <console-input> line 1:
  (source code not available)

A managed resource "azurerm_resource_group" "test" has not been declared in
the root module

>  
```

The way errors render for console input is still a little odd here since we don't have the source code snippet available, but we'll address that later as a separate issue.

Since this fix is already in master ready to be included in the final v0.12.0 release, I'm going to close this out now. Thanks for reporting it, @tombuildsstuff!
